pages/mesosphere/dcos/2.2/storage/external-storage/csi/index.md

```diff
@@ -0,0 +1,117 @@
+
+## Create an application from the DC/OS UI
+
+1. Click the **Services** tab, then **RUN A SERVICE**.
+1. Click **Volumes** and enter your Volume ID, Plugin Name, and Container Path.
+1. Click **Deploy**.
+
+# Scaling your app
+
+Marathon apps which use CSI volumes can currently only be scaled to a single instance. This constraint will be relaxed in the future.
+
+# Data mobility
+
+If a Marathon application using a CSI volume fails for some reason, Marathon will relaunch that service wherever available resources are found. Since CSI volumes may be attached to any one of the nodes in the cluster running the CSI plugin, Marathon may place it on a different physical host. This is one major advantage of using external storage with your applications: data mobility means that simple, non-distributed stateful applications can tolerate failure with only a short period of downtime.
+
+# Using 3rd party CSI plugins
+
+If you want to use a third party CSI plugin which has not yet integrated with DC/OS, you can do the following to accomplish this:
+
+1. Install the CSI plugin as a Marathon app. To integrate with DC/OS, the CSI plugin must expose a Unix domain socket which DC/OS can connect to in order to make CSI calls. Install the app with a `hostname:UNIQUE` constraint so that only one instance is installed on each node. To make this storage backend available on all nodes, install the plugin on all private agents by setting the number of instances equal to the number of private agents in the cluster.
+
+2. Place a configuration file on each private agent in the folder `/opt/mesosphere/etc/dcos/storage/csi/` which looks like the following:
+
+```json
+{
+  "type": "test.pluginname.com",
+  "endpoints": [
+    {
+      "csi_service": "NODE_SERVICE",
+      "endpoint": "unix:///var/lib/pluginname/csi.sock"
+    }
+  ],
+  "target_path_root": "/var/lib/pluginname/target_root"
+}
+```
+
+The `target_path_root` parameter specifies the root directory, managed by the CSI plugin, where mount targets will be created. Once the app is successfully running on all agents, the plugin is ready for you to use it in DC/OS. The `type` parameter contains the unique name of the CSI plugin which should be passed in the `pluginName` field of the external volume in your Marathon app definitions.
+
+Note that the current implementation of CSI support in DC/OS only makes calls against the [node service](https://github.com/container-storage-interface/spec/blob/master/spec.md#node-service-rpc) of the plugin; this means that any plugins which require calls against the controller service are not currently supported. Support for such plugins is planned in the future.
+
+# Potential issues
```

Comment:
Do you want to mention here that only MESOS container runtime is supported for CSI but not DOCKER runtime?